include/fiction/algorithms/network_transformation/inverter_balancing.hpp

```diff
@@ -0,0 +1,323 @@
+            [this, &ntk_dest, &old2new](const auto& po)
+            {
+                auto tgt_signal = old2new[ntk.get_node(po)];
+                auto tgt_po     = ntk.is_complemented(po) ? ntk_dest.create_not(tgt_signal) : tgt_signal;
+                if(tgt_po == 0)
+                {
+                    tgt_signal = old2new[ntk.get_node(preserved_po[0])];
+                    preserved_po.erase(preserved_po.begin());
+                    tgt_po     = ntk.is_complemented(po) ? ntk_dest.create_not(tgt_signal) : tgt_signal;
+                }
+                ntk_dest.create_po(tgt_po);
+            });
+
+        if constexpr (mockturtle::has_foreach_ri_v<TopoNtkSrc>){
+            ntk.foreach_ri(
+                [this, &ntk_dest, &old2new](const auto& po)
+                {
+                    const auto tgt_signal = old2new[ntk.get_node(po)];
+                    const auto tgt_po     = ntk.is_complemented(po) ? ntk_dest.create_not(tgt_signal) : tgt_signal;
+
+                    ntk_dest.create_ri(tgt_po);
+                });
+        }
+
+        // restore signal names if applicable
+        fiction::restore_names(ntk, ntk_dest, old2new);
+
+
+        return ntk_dest;
+
+    }
+
+  private:
+    using TopoNtkSrc = mockturtle::topo_view<NtkSrc>;
+    TopoNtkSrc ntk;
+    mockturtle::fanout_view<TopoNtkSrc> fo_ntk{ntk};
+    using node = typename NtkSrc::node;
+    std::vector<node> del_inv;
+    std::vector<node> new_inv;
+    std::vector<node> blc_fos;
+    std::vector<node> preserved_po;
+};
+
+template<typename NtkSrc>
+bool inverter_balancing_recursive(const NtkSrc& ntk){
+    bool return_val = false;
+    const auto fo_ntk= mockturtle::fanout_view<NtkSrc>(ntk);
+    ntk.foreach_gate(
+        [&](const auto& g)
+        {
+            if (ntk.fanout_size(g)>=2)
+            {
+                auto fanout_inv = fanouts(fo_ntk, g);
+                auto balance = std::all_of(fanout_inv.begin(), fanout_inv.end(),[&](const auto& fo_node)
+                                              {return fo_ntk.is_inv(fo_node);});
+                if(balance && fanout_inv.size()>1){
+                    return_val = true;
+                }
+            }
+        });
+    return return_val;
+}
+
+}  // namespace detail
+
+/**
+ * Converts a logic network into an equivalent one of another type. Thereby, this function is very similar to
+ * mockturtle::cleanup_dangling. However, it supports real buffer nodes used for fanouts and path balancing in fiction.
+ *
+ * When both fanouts of a fanout-node are inverted, the inverters are replaced by one inverter as fanin of this node
+ * This is part of the Distribution Newtork I: Inputs
+ *
+ *
+ * @tparam NtkDest Type of the returned logic network.
+ * @tparam NtkSrc Type of the input logic network.
+ * @param ntk The input logic network.
+ * @return A logic network of type NtkDest that is logically equivalent to ntk.*/
+
+
+template <typename NtkSrc>
+NtkSrc inverter_balancing(const NtkSrc& ntk)
```

Comment:
While it's not a bad name, it might be a bit long to type. We want to have as short, but still descriptive, commands as possible. Maby something like just `inv` could be possible? We can then later also extend the command with (more) flags in case we develop other inverter-based techniques.